File: Back/.Net/BetFriends.Domain/Entity.cs
namespace BetFriends.Domain;

public abstract class Entity
{
    private readonly List<IDomainEvent> events;
    public IReadOnlyCollection<IDomainEvent> Events { get => events.AsReadOnly(); }

    public Entity()
    {
        events = [];
    }

    protected void AddEvent(IDomainEvent @event)
    {
        events.Add(@event);
    }
}

File: Back/NodeJS/modules/shared/domain/Entity.ts
import { IDomainEvent } from "./IDomainEvent";

export abstract class Entity {
    private domainEvents: IDomainEvent[] = [];
    public DomainEvents: IDomainEvent[] = this.domainEvents;

    protected AddDomainEvent(event: IDomainEvent) {
        this.domainEvents.push(event);
    }
}